test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -707,15 +712,15 @@ private String buildNodeName(int id, Settings settings) {
      */
     private static String getRoleSuffix(Settings settings) {
         String suffix = "";
-        if (Node.NODE_MASTER_SETTING.exists(settings) && Node.NODE_MASTER_SETTING.get(settings)) {
+        if (settings.hasValue("node.roles") && DiscoveryNode.hasRole(settings, DiscoveryNodeRole.MASTER_ROLE)) {
```

Comment:
The method is testing `settings.hasValue("node.roles")` three times. Could we just test it once, and put the rest of the code in the "then" block?

Reply:
Thanks, I included this.